Sources/ComposableArchitecture/SwiftUI/Binding.swift

```diff
@@ -258,4 +258,29 @@ extension ViewStore {
       send: { action(.set(keyPath, $0)) }
     )
   }
+
+  /// Derives a binding from the store that mutates state by wrapping a ``BindingAction`` with the
+  /// store's action type.
+  ///
+  /// For example, a text field binding can be created like this:
+  ///
+  /// ```swift
+  /// struct State { var text = "" }
+  /// enum Action { case binding(BindingAction<State>) }
+  ///
+  /// TextField(
+  ///   "Enter text",
+  ///   text: viewStore.binding(Action.binding).text
+  /// )
+  /// ```
+  ///
+  /// - Parameter action: A function that wraps a binding action in the view store's action type.
+  /// - Returns: A binding.
+  public func binding(_ action: @escaping (BindingAction<State>) -> Action) -> Binding<State>
+  where State: Equatable {
```

Comment:
All of state needs to be equatable to return a `Binding` here. This is a common enough practice in TCA apps, but an obstacle nonetheless.

We could overcome that by having a `ViewStore.Binding` helper that has dynamic member lookup over equatable sub-state, instead, but wanted to start a convo first to see if it's worth introducing that type just to clean up call sites a lil bit.

Reply:
This is already the case with the current `.binding(keyPath:send:)` right?

Reply:
Nah that's only equatable on the `Value`. This requires equatable on `Root`.